src/qengine/state/state.cpp

```diff
@@ -87,28 +87,47 @@ void QEngineCPU::SetQuantumState(Complex16* inputState)
  *
  * A fundamental operation used by almost all gates.
  */
-void QEngineCPU::Apply2x2(bitCapInt offset1, bitCapInt offset2, const Complex16* mtrx, const bitLenInt bitCount,
+void QEngineCPU::Apply2x2(bitCapInt offset1, bitCapInt offset2, const complex* mtrx, const bitLenInt bitCount,
     const bitCapInt* qPowersSorted, bool doCalcNorm)
 {
-    Complex16 nrm = Complex16((bitCount == 1) ? (1.0 / runningNorm) : 1.0, 0.0);
     int numCores = GetConcurrencyLevel();
+    double nrm = 1.0 / runningNorm; 
+#if ENABLE_AVX  
+    ComplexUnion mtrxCol1;
+    ComplexUnion mtrxCol2;
+    mtrxCol1.cmplx[0] = mtrx[0];
```

Comment:
If you’re going to have them, you might as well use the ComplexUnion constructors.

Reply:
Good point.